cfme/common/__init__.py

```diff
@@ -155,6 +161,135 @@ def get_tags_db(self):
+        """ Add tag to tested item
+            Args:
+                category: category(str)
+                tag: tag(str) or Tag object
+                cancel: set True to cancel tag assigment
+                reset: set True to reset already set up tag
+        """
+        view = navigate_to(self, 'EditTags')
+        if isinstance(tag, Tag):
+            category = tag.category.display_name
+            tag = tag.display_name
+        try:
+            view.fill({
+                "tag_category": '{} *'.format(category),
+                "tag_name": tag
+            })
+        except NoSuchElementException:
+            view.fill({
+                "tag_category": category,
+                "tag_name": tag
+            })
+        self._tags_action(view, cancel, reset)
+
+    def edit_tags(self, tags):
+        """Add multiple tags
+            Args:
+                tags: pass dict with category name as key, and tag as value,
+                     or pass list with tag objects
+        """
+        if isinstance(tags, dict):
+            for category, tag in tags.items():
+                self.edit_tag(category=category, tag=tag)
+        elif isinstance(tags, (list, tuple)):
+            for tag in tags:
+                self.edit_tag(tag=tag)
+
+    def remove_tag(self, category=None, tag=None, cancel=False, reset=False):
+        """ Remove tag of tested item
+               Args:
+                   category: category(str)
+                   tag: tag(str) or Tag object
+                   cancel: set True to cancel tag deletion
+                   reset: set True to reset tag changes
+               """
+        view = navigate_to(self, 'EditTags')
+        if isinstance(tag, Tag):
+            category = tag.category.display_name
+            tag = tag.display_name
+        try:
+            row = view.tag_table.row(category="{} *".format(category), assigned_value=tag)
+        except Exception:
+            row = view.tag_table.row(category=category, assigned_value=tag)
+        row[0].click()
+        self._tags_action(view, cancel, reset)
+
+    def remove_tags(self, tags):
+        """Remove multiple of tags
+            Args:
+                tags: pass dict with category name as key, and tag as value,
+                     or pass list with tag objects
+        """
+        if isinstance(tags, dict):
+            for category, tag in tags.items():
+                self.remove_tag(category=category, tag=tag)
+        elif isinstance(tags, (list, tuple)):
+            for tag in tags:
+                self.remove_tag(tag=tag)
+
+    def get_tags(self, tenant="My Company Tags"):
+        """ Get list of tags assigned to item
+            Args:
+                tenant: string, tags tenant, default is "My Company Tags"
+            Returns:
+                List of tags in format "Tag_category: Tag_name"
+        """
+        view = navigate_to(self, 'Details')
+        tags = view.tag.read()
```

Comment:
yes, as this tag summary widget can be different ex. for group tag with use SummaryFormItem, and for vm/instances another one, so it better to add it to Details view